pkg/flowkit/account.go

```diff
@@ -28,11 +28,82 @@ import (
 	"github.com/onflow/flow-cli/pkg/flowkit/util"
 )
 
-// Account is a flowkit specific account implementation.
+// Account is a flowkit-specific account implementation.
 type Account struct {
 	name    string
 	address flow.Address
 	key     AccountKey
+
+	// fromFile is the configuration file containing this account.
+	//
+	// This field is only set if the external "fromFile"
+	// syntax is used. Otherwise this field is empty.
+	//
+	// Ref: https://docs.onflow.org/flow-cli/security/#private-account-configuration-file
+	fromFile string
+	// useAdvancedSaveFormat is to save an account in advanced format even if it has default values as defined
+	// in isDefaultKeyFormat() in flowkit/config/json/account.go
+	// defaults to false
+	useAdvancedSaveFormat bool
```

Comment:
When reviewing the UX, we thought that the contents of the `{name}.private.json` file were hard to understand for newcomers to Flow. In particular, it's not clear that the single "key" field is a private key. The implicit defaults also (ECDSA_P256, SHA3_256) aren't obvious. So we wanted to save the account in advanced format to make those things clear (we can revisit this decision, curious to get your opinion). I think the convenience is worth it on the emulator, but for production mainnet keys, the explicit format feels safer. 

When implementing that, we also noticed that if a user manually adds a key with default values in advanced format, the config module will re-save it in simplified format. I'm not sure if that's intentional, but it felt undesirable to automatically reformat a user's key.

All that said, I agree that this crosses domain boundaries. Maybe we don't need this, but if we do, I'm a bit stuck on how to implement it cleanly.

Reply:
> When implementing that, we also noticed that if a user manually adds a key with default values in advanced format, the config module will re-save it in simplified format. I'm not sure if that's intentional, but it felt undesirable to automatically reformat a user's key.

That was a conscious change, but I was on the fence I remember. I decided to do it that way to really migrate users to the new format. 

In my mind, we don't need to save default values, since we are consistently not doing that anymore, but I really feel this is subjective, so I don't want to push it. In my mind, if a user is during the process explained what the algos were he should understand that, but I even think that users won't really understand this, since they are using the "beginner friendly" method for account creation. 

If we decide to keep it we can implement it along with the change I suggested about saving to a separate file. That one is https://github.com/onflow/flow-cli/pull/538#discussion_r921993732 and I think that suggestion can work for this too, because state will know which accounts is saving separately and it can set that property on the config/account (which is fine if it's there) but I would remove it on flowkit/account.